### ISSUE TITLE
Selected runs are not logging media for the key: score per update, but instead are logging type

### ISSUE THREAD

**Issue by Kimonili** [wandb/client#1206]
* Weights and Biases version: 0.9.5
* Python version:  3.8.5
* Operating System: Windows 10

### Description

I was trying to log some values using `wandb.log()` but got one of the windows of my logs said this instead as shown in the attached file.
![error](https://user-images.githubusercontent.com/59944048/90507212-30cae480-e15e-11ea-8080-5c9abb2a71ee.png)
The other logs seems to work fine except the one of `np.mean(total_score_per_update)`
![error1](https://user-images.githubusercontent.com/59944048/90507754-18a79500-e15f-11ea-891f-61cab3c476c9.png)

### What I Did

```
            for self.step in range(0, num_steps):
                self.states[self.step] = next_state
                self.masks[self.step] = next_done
                # given a state agent selects an action
                with torch.no_grad():
                    next_state, reward, done, _ = self.select_make_action(self.states[self.step])
                    self.rewards[self.step] = reward
                    next_done = torch.Tensor(done).to(self.device)
                    next_state = torch.Tensor(next_state).to(self.device)
                if self.step == 0:
                    score = reward
                else:
                    score += reward
                total_score_per_update.append(score)
                score_avg.append(score)
                if len(score_avg) == score_threshold:
                    if np.mean(score_avg) > -0.5:
                        print("Environment is solved and achieved an average of {} for the last {} episodes"
                              .format(np.mean(score_avg), len(score_avg)))
                        end_time = time()
                        print("It took {} seconds to solve the environment"
                              .format(end_time - start_time))
                        break
            if len(score_avg) == score_threshold:
                if np.mean(score_avg) > -0.5:
                    break
            actor_loss, critic_loss = self.update_model(next_state)
            wandb.log({"actor loss": actor_loss, "critic loss": critic_loss,
                       "actor learning rate": actor_lr_now,
                       "critic learning rate": critic_lr_now, 
                       "score per update": np.mean(total_score_per_update)})

```

**Comment by vanpelt** [comment]
Can you try calling `np.mean(total_score_per_update).item()`, I think it's logging an array of values.

**Comment by Kimonili** [comment]
I did that, but still the same message occurs. I do not know if it is helpful, but `type(np.mean(total_score_per_update).item())` returns float, so probably the type of the logged variable is not the cause of the problem.

Might this happen because `np.mean(total_score_per_update)` is calculated outiside the nested for loop (for every `update`), although the `score` (for every `self.step`) is appended to `total_score_per_update` deque inside the nested for loop? Is there a possibility for this to cause a "collision"?

**Comment by vanpelt** [comment]
Can you send me a link to a run where this is happening?

**Comment by Kimonili** [comment]
Yes sure! Here you go: https://app.wandb.ai/k_ili/multiagent-particle-envs/runs/3thu4nzq?workspace=user-k_ili

**Comment by vanpelt** [comment]
Hey @Kimonili, it looks like originally that chart got added as a "Media Panel" likely because it was logging an array.  You can reset your workspace (by clicking this [link](https://app.wandb.ai/k_ili/multiagent-particle-envs/runs/3thu4nzq?workspace=empty)) which will pickup the "score per update" chart, or manually add a new chart and choose "score per update" as the y-axis.

**Comment by AvivSham** [comment]
Hi all,
I'm still facing the same issue I log two different numpy arrays in my code, one of them I can see over the dashboard and the other one I can't. I also tried the trick with ```.item()``` but it didn't work for me. 
I spoke with Diganta about this issue and he told me he would try to reproduce it.

**Comment by AvivSham** [comment]
It looks like it is logged as a media plot and not as a line plot. The workaround is to create a new panel with Y-axis of the desired variable. I must say it's not convenient and it's worth investigating why does is happen.

**Comment by ariG23498** [comment]
Hey @AvivSham 
I tried reproducing the ticket and observed the following :point_down: 
## Code that I used
I tried logging three runs in my project
1. `numpy` arrays
```python
a_numpy = np.array([1,2,3,4,5])
b_numpy = np.array([10,20,30,40,50])
with wandb.init(entity='xxx', project='xxx', name='numpy_test') as run:
    run.log({'a_numpy':a_numpy, 'b_numpy':b_numpy})
```
2. default `list`
```python
a_list = [1,2,3,4,5]
b_list = [10,20,30,40,50]
with wandb.init(entity='xxx', project='xxx', name='list_test') as run:
    run.log({'a_list':a_list, 'b_list':b_list})
``` 
3. logging the values iteratively
```python
a_list = [1,2,3,4,5]
b_list = [10,20,30,40,50]
with wandb.init(entity='xxx', project='xxx', name='iterative_test') as run:
    for a,b in zip(a_list,b_list):
        run.log({'a':a, 'b':b})
```
## A quick insight into summary metrics:
According to the [documentation of wandb](https://docs.wandb.ai/library/log#summary-metrics), there is something known as the summary metrics. The summary statistics are used to track single metrics per model. Summary can handle `numpy` arrays, `pytorch tensors` or `tensorflow tensors`. When a user `logs` lists or tensors, the values are logged as a summary metric. This can be seen in the overview page of the runs.
**Numpy_test**
![image](https://user-images.githubusercontent.com/36856589/106236044-0c515900-6222-11eb-983d-c7f3f8a7673e.png)
**List_test**
![image](https://user-images.githubusercontent.com/36856589/106236078-1ecb9280-6222-11eb-8be3-984723c776e2.png)

wandb internally suggests these as media and tries plotting them with media panels (which should not be the case).

On the other hand with iterative logging, the `step` parameter of the `log` method increments iteratively while logging the values that we want. Drawing a parallel here with summary metrics, the iterative logging treats the last value as the summary metric as can be seen here :point_down: 
![image](https://user-images.githubusercontent.com/36856589/106236361-b4ffb880-6222-11eb-9769-888679d34fd1.png)

Let me know if there is anything else that I missed out on the reproduction. I think this issue reproduction along with the sleuthing is worth a lot to the core team to fix the discrepancies. :detective:

**Comment by cvphelps** [comment]
@AvivSham I would recommend using Custom Charts if you have a table of values you're logging, instead of individual scalars per step: https://docs.wandb.ai/app/features/custom-charts

**Comment by davidcian** [comment]
I've also had the same problem. I think the issue might be that if you log a NumPy array with a certain key, then start logging a scalar value with the *same* key, wandb doesn't automatically start plotting it on the dashboard as if it had been a scalar from the very beginning. As mentioned above, adding the graph manually works.

**Comment by ariG23498** [comment]
Closing this issue as the workaround suggested by @cvphelps works in these scenarios!

**Comment by richardrl** [comment]
> Hey @Kimonili, it looks like originally that chart got added as a "Media Panel" likely because it was logging an array. You can reset your workspace (by clicking this [link](https://app.wandb.ai/k_ili/multiagent-particle-envs/runs/3thu4nzq?workspace=empty)) which will pickup the "score per update" chart, or manually add a new chart and choose "score per update" as the y-axis.

This scenario happens a lot. When we are adding a new metric, it might be accidentally logged as a list and show up as Media Panel. Is there any way to automatically reset the workspace? Or have some UI element? I forget this trick of workspace=empty

**Comment by CiaoHe** [comment]
I think the problem is: that you have to empty your workspace (I just delete the whole project) once you made such mistakes(log a NumPy array at one time). It really destroys me

**Comment by ZhongxuanWang** [comment]
> I think the problem is: that you have to empty your workspace (I just delete the whole project) once you made such mistakes(log a NumPy array at one time). It really destroys me

THANK YOU!!! I'VE WASTED 1.5h ON THIS...

**Comment by botcs** [comment]
@CiaoHe thanks that actually works.
Unfortunately just removing the single run where you made the mistake is not enough...

I'll try to raise an issue
